### DISCUSSION TITLE
Use a single ResizeObserver instance

### DISCUSSION THREAD

**Discussion by ZeeCoder** [ZeeCoder/use-resize-observer#80] [Ideas]
Seems like it's more performant and the recommended pattern:
https://github.com/WICG/ResizeObserver/issues/59

**Comment by shiba-codes** [comment]
Related – https://github.com/ZeeCoder/use-resize-observer/pull/26

**Comment by ZeeCoder** [comment]
I'm not convinced yet whether this is an optimisation that's actually worth it or not.
I've only seen a single benchmark that seem to indicate that it might be better to use a single RO for all hook instances,
but it wasn't convincing enough to warrant the associated work.

I'd need a way to test this somehow for myself for a large number of observed elements and measure CPU / memory usage maybe, to see if there's a difference between using one or many RO instances.

**Comment by ZeeCoder** [comment]
Or if there was a comprehensive test / article / benchmark that demonstrated that using a single RO instance is significantly better, then I would consider it.

**Comment by ZeeCoder** [comment]
I forgot about this one:
https://github.com/que-etc/resize-observer-polyfill/issues/13#issuecomment-304648939

Seems like there's something in there, but I'd still need a way to benchmark CPU + memory usage, so that I could compare a results of the current implementation to one that uses a single instance.

**Comment by ZeeCoder** [comment]
I'd really appreciate someone well-versed in benchmarking UI to help out.

**Comment by codingedgar** [comment]
The implementation of https://github.com/jaredLunde/react-hook/blob/1dd3940a70699836347325dac2cfe9998b81a5a4/packages/resize-observer/src/index.tsx might work as inspiration.

**Reply by ZeeCoder** [reply]
yeah that's pretty much the approach I'd take.
Only with WeakMap instead of Map.

**Comment by ZeeCoder** [comment]
The supposed performance benefits of this don't seem to be real anymore, so I'm closing this.
See these AI findings: https://gist.github.com/ZeeCoder/169365d5cf424837a660a2d33478b6de
